apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/os/OSUtil.java

```diff
@@ -50,6 +50,13 @@ public static String getHostName() {
                 HOST_NAME = host.getHostName();
             } catch (UnknownHostException e) {
                 HOST_NAME = "unknown";
+                String host = e.getMessage();
```

Comment:
According to `UnknownHostException`, `Thrown to indicate that the IP address of a host could not be determined.`.

Could you explain why you need this? Once you set hostname <-> ip address mapping in your OS, you should be fine.

Reply:
The original discussion seems very clear about the solution, https://github.com/apache/skywalking/issues/4520#issuecomment-599959270. I am not sure why you are asking for code change? And at the same time, you are reading a hostname based on exception message, clearly, this is not following the JDK codes or spec. You could find many `throw new UnknownHostException(...)` in the JDK source codes w/o ip and `:`.

Reply:
In #4520, say it‘s a special case.
But i Think this following scenarios are many,
only change hostname, like `FooBar-01` in `/etc/hostname`(in linux),
usually, we don't add `192.xxx.xxx.xxx FooBar-01` in `/etc/hosts` or add it in our private dns service.
I aggre you say in #4520, 
but in this situation, agent get hostname is `FooBar-01` better than `unknown`.

Reply:
> But i Think this following scenarios are many,
only change hostname, like FooBar-01 in /etc/hostname(in linux),
usually, we don't add 192.xxx.xxx.xxx FooBar-01 in /etc/hosts or add it in our private dns service.

Actually, special case means, this could be done in a standard way. Rather than changing codes. 
And please read the second half of my comments, the way you are using to process the exception message, is really not an official way.